activerecord/test/models/company.rb

```diff
@@ -26,6 +26,9 @@ def arbitrary_method
   def private_method
     "I am Jack's innermost fears and aspirations"
   end
+
+  class SpecialCo < Company
+  end
```

Comment:
@agrobbin seems like a bug :sweat:. You could try one of the other subclasses in the `Namespaced` module to get this fix out the door. Would you be interested to look at the issue with `find_sti_class` in combination with `Firm`?